server/api/notifications.py

```diff
@@ -0,0 +1,131 @@
+from sqlalchemy.orm.exc import NoResultFound
+from flask import Blueprint as bp
+from flask import (
+    make_response,
+    jsonify, request,
+    session, g
+)
+from auth import authenticate
+from models.user import User
+from models.notification import Notification
+
+notif_views = Blueprint("notifications", __name__, url_prefix="notifications")
+
+@notif_views.route("/mark_read")
+@authenticate
+def mark_read(**kwargs):
+    """
+    Mark a notification as read
+    """
+    response_dict = {}
+    # Probably needs to be json
+    notif_id =  request.get_json().get('id', None)
+    # Try to get the notification id
+    if notif_id == None:
+        response_dict['status'] = 417
+        response_dict['message'] = "Failed to get notification id"
+        return jsonify(response_dict), 417
+    # Try to get the notification object
+    try:
+        notification = Notification.get_instance(**{'id': notif_id})
+    except NoResultFound:
+        response_dict['status'] = 404
+        response_dict['message'] = "Notification with id %s was not found" % (notif_id)
+    Notification.update(notification.id, **{'read': True})
+    # Return an updated list of notifications
+    response_dict['notifications'] = []
+    for notification in notifications:
+        response_dict['notifications'].append(
+            notification.to_dict(excludes=['user', 'user_id'])
+        )
+    response_dict['status'] = 200
+    return jsonify(response_dict), 200
+
+
+
+@notif_views.route("/create_notification")
+@authenticate
+def create_notification(**kwargs):
+    """
+    Create a new notification and link it
+    with the logged in user
+    """
```

Comment:
I'm not clear what the use case of this would be, all of the user's notifications would originate from another account